### DISCUSSION TITLE
change the carrier board and device tree for Jetson Nano

### DISCUSSION THREAD

**Discussion by danny1209** [OE4T/tegra-demo-distro#205] [Q&A]
Hi all, 
this repository works fine for my standard jetson nano devkit from nvidia.
now I change the carrier board, that means the corresponding device tree must also be changed. 
I find the device tree are in this folder:
/../tegra-demo-distro/build/tmp/work-shared/jetson-nano-devkit-emmc/kernel-source/nvidia/platform/t210/porg/kernel-dts 

so I replace the files here with my own device tree files, espacially the file "tegra210-p3448-0002-p3449-0000-b00.dts"
Everytime when i rerun the command "bitbake demo-image-full", it just skiped all the changed files. 
therefore I want to know how to only rebuild and install this part in image? thanks for any help.

Best regard
Daniel

**Comment by ichergui** [comment]
Hello @danny1209 
Did you add the rule in the `Makefile` to generate the `.dtb` file ?
Check first in the `linux-tegra` work directory if you are able to see your new `dtb` file

**Reply by danny1209** [reply]
hi @ichergui 
yes I have Makefile, but no new dtb files are generated.

**Reply by danny1209** [reply]
I used "bitbake -c cleansstate virtual/kernel" to remove the existed kernel. 
after that I changed the files in /tegra-demo-distro/build/workspace/sources/linux-tegra/nvidia/platform/t210/porg/kernel-dts
and rebuild it with bitbake demo-image-full, but in images I only get a part of the files. original was 43, now only 17
![image](https://user-images.githubusercontent.com/46848895/166725750-cd7b3bfa-4600-4fbb-bbfa-31537edb56e4.png)
of course without this file and failed.
![image](https://user-images.githubusercontent.com/46848895/166725929-07c3cb67-9a20-4102-beba-e5523145abba.png)

any idea?

**Reply by dwalkes** [reply]
> after that I changed the files in /tegra-demo-distro/build/workspace/sources/linux-tegra/nvidia/platform/t210/porg/kernel-dts

I don't think you'll want to hack on that content directly.  I'd suggest something like the responses in https://stackoverflow.com/questions/68222619/update-custom-device-tree-on-yocto using a bbappend on the `linux-tegra` kernel recipe.  Ridgerun also has a page at https://developer.ridgerun.com/wiki/index.php?title=Yocto_Support_for_NVIDIA_Jetson_Platforms_-_Additional_Topics but it looks a bit out of date with the devicetree paths.

Regarding your test here I suspect you can't just rename the previous `build/tmp/work-shared/jetson-nano-devkit-emmc` directory and get the same content there with a new build after cleansstate, if this is what you did.

**Reply by danny1209** [reply]
hi @dwalkes,
thanks for you reply. 
the first suggestion seems very complicated. 
I tried the second suggestion with devtool modify virtual/kernel, and change the source code in workplace, not in the tmp directory. 
Regarding the names, I just wanted to save the old imagefiles and compare them with the new generated directory. 
Therefore I found the difference, rebuild just generated only part of the imagefiles, which are genereted from the first time. 
also because of this, the final image can not be created successfully.

**Reply by danny1209** [reply]
Update: it works!
the original image folder "jetson-nano-devkit-emmc" should not be renamed.
